pkg/dataplane/reloadCredentials.go

```diff
@@ -0,0 +1,189 @@
+package dataplane
+
+import (
+	"context"
+	"log/slog"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/msi-dataplane/pkg/dataplane/swagger"
+	"github.com/fsnotify/fsnotify"
+	"github.com/go-logr/logr"
+)
+
+const (
+	// Errors returned when reloading credentials
+	errCreateFileWatcher = "failed to create file watcher"
+	errAddFileToWatcher  = "failed to add credentialFile to file watcher"
+	errLoadCredentials   = "failed to load credentials from file"
+)
+
+type ReloadingCredential struct {
+	ClientCertificateCredential *ClientCertificateCredential
```

Comment:
Either unexport this type or store the two fields separately inside of `ReloadingCredential`, user should never know about this implementation detail.